src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.AnalyzerExecutionContext.cs

```diff
@@ -17,6 +17,11 @@ internal partial class AnalyzerManager
     {
         private sealed class AnalyzerExecutionContext
         {
+            /// <summary>
+            /// Cached mapping of localizable strings in this this descriptor to any exceptions thrown while obtaining them.
+            /// </summary>
+            private static ImmutableDictionary<LocalizableString, Exception?> s_localizableStringToException = ImmutableDictionary<LocalizableString, Exception?>.Empty.WithComparers(Roslyn.Utilities.ReferenceEqualityComparer.Instance);
```

Comment:
❓  Can we use `ImmutableSegmentedDictionary` here? The storage overhead is reduced relative to `ImmutableDictionary`, and the lookup performance (which is the primary fast path) is significantly improved.

Reply:
Yup yup. I'll make that change.